text/0001-catalogs.md

```diff
@@ -0,0 +1,341 @@
+
+- These giant edits lead to churn in `package.json` files and merge conflicts with other commits editing `package.json` files.
+- Repository maintainers have to remember to run this synchronization step periodically, or create a Continuous Integration step to verify the repo is in a good state.
+
+Neither of these error-prone operations are necessary when using a catalog.
+
+### 2. Intention becomes clear
+
+There might be a tight relationship between `foo` and `bar`.
+
+```json5
+{
+  "name": "@monorepo/foo",
+  "dependencies": {
+    "react": "^17.0.2"
+  }
+}
+```
+
+```json5
+{
+  "name": "@monorepo/bar",
+  "dependencies": {
+    "react": "^17.0.2"
+  }
+}
+```
+
+A developer working primarily in `@monorepo/bar` may not realize the implied coupling and upgrade `@monorepo/bar` to `react@18` without realizing an edit to `@monorepo/foo` was also required.
+
+The `catalog:` protocol makes it more clear from just reading `package.json` when a dependency is intended to be consistent across the monorepo. Ideally this person would search "catalog package.json` online and find pnpm.io docs.
+
+## Implementation
+
+### Fetching
+
+Catalogs can be shared through standard NPM packages. The `pnpm-catalog-v1` key in `exports` will be used when referencing an external catalog.
+
+```json
+{
+  "name": "@example/frontend-catalog",
+  "exports": {
+    "pnpm-catalog-v1": "./catalog.yaml"
+  }
+}
+```
+
+The schema of a catalog may be extended in the future. In this hypothetical, the `-v1` portion of the exports key enables a package to publish catalogs for multiple versions of pnpm that understand different schema versions.
+
+Only the subset of the version catalog that's used will be fetched and installed.
+
+### Lockfile
+
+Catalogs will be saved to `pnpm-lock.yaml` under a new `catalogs` key. This allows users to more easily review changes to catalogs and pnpm to perform faster up-to-date checks.
+
+```yaml
+lockfileVersion: next
+
+importers:
+
+  packages/foo:
+    dependencies:
+      react:
+        specifier: 'catalog:@example/frontend-catalog'
+        version: ^18.2.0
+
+catalogs:
+
+  /@example/frontend-catalog@0.1.0:
+    resolution: {integrity: sha512...}
+    versions:
+      react:
+        specifier: ^18.2.0
+      react-dom:
+        specifier: ^18.2.0
+      redux:
+        specifier: ^4.2.0
+      react-redux:
+        specifier: ^8.0.0
+
+packages:
+  # ...
+```
+
+### Other Changes
+
+Similar to the [`workspace:` protocol](https://pnpm.io/workspaces#workspace-protocol-workspace), `pnpm publish` will need to replace instances of `catalog:` with valid specifiers before publishing.
+
+### Add and update commands
+
+The `pnpm add` command will add versions from default catalog if it's configured. The `pnpm update` command will prompt users if they wish to update specifiers in a catalog.
+
+## Extensions
+
+### Merge Conflict Resistance in the Lockfile
```

Comment:
I do have some ideas on how we can significantly reduce lockfile merge conflicts, but happy punt this to a followup RFC or discussion. Last time this was discussed, the peer dependencies suffix made this complicated. My thoughts aren't 100% complete yet, so I'll avoid commenting ideas until that's ready.